packages/theme/src/cli/services/open.ts

```diff
@@ -10,17 +10,16 @@ export async function open(
   adminSession: AdminSession,
   options: {development: boolean; live: boolean; editor: boolean; theme: string | undefined},
 ) {
-  let themeId = options.theme
-  if (options.development) {
-    const theme = await new DevelopmentThemeManager(adminSession).find()
-    themeId = theme.id.toString()
-  }
-
+  const developmentThemeManager = new DevelopmentThemeManager(adminSession)
+  const developmentTheme = (
+    await (options.development ? developmentThemeManager.find() : developmentThemeManager.fetch())
```

Comment:
this branching logic to decide which method to use on the `developmentThemeManager` is appearing a lot, which feels a bit like a leaky abstraction 🤔 do you think it's possible and sensible to move this logic into the development theme manager?

Reply:
yehhhh i see what you're saying, the thing that feels different in this scenario is the call site is deciding which method to use on DevelopmentManager based on an option/flag that could just be passed to the constructor. and every case ive seen the default is to use fetch() if development id isn't set

im cool to leave as is tho 👍